src/components/solvers/SolverPicker.tsx

```diff
@@ -0,0 +1,57 @@
+import React, {useEffect, useState} from "react";
+import {Button, Container, Icon, ListItem, Text, UnorderedList } from "@chakra-ui/react";
+import {fetchSolvers} from "../../api/ToolboxAPI";
+import {ProblemSolver} from "./ProblemSolver";
+import {InputButton} from "./buttons/InputButton";
+import {TbBulb} from "react-icons/tb";
+
+export interface SolverPickerProps {
+    problemType: string;
+    onSolverPicked?: (solver: ProblemSolver | null) => void;
+}
+
+export const SolverPicker = (props: SolverPickerProps) => {
+    const [loadingSolvers, setLoadingSolvers] = useState<boolean>(true);
+    const [solvers, setSolvers] = useState<ProblemSolver[]>([]);
+
+    useEffect(() => {
+        setLoadingSolvers(true);
+        fetchSolvers(props.problemType)
+            .then(solvers => {
+                setSolvers(solvers);
+                setLoadingSolvers(false);
+            })
+    }, [props.problemType]);
```

Comment:
What is this effect used for? The problem type shouldn't change or does it?

Reply:
It's used for the initial loading process. If there is a better way to do it, please let me know. If anything I can definitely restrict it to only load once.
I didn't see any hook that runs once in a const component. Only `componentDidMount()` in a class based react component.

Reply:
My bad, this is the correct place 👍